contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,179 @@
+
+### Angular when compared with other popular frontend frameworks
+
+
+There are many frontend frameworks out that many developers prefer over Angular. Two of the most popular tools are <a href="https://reactjs.org/">React</a> and <a href="https://vuejs.org/">Vue</a>.
+
+Choosing a tech stack sometimes becomes a tedious task as you need to take every factor into consideration including budget, time, app size, end-users, project objectives, and resources.
+
+Angular is opinionated and tells developers how to structure their code. It greatly eases the engineering process, especially in large teams. However the steep learning curve, amongst other disadvantages,
+make developers adopt other tools.
+
+React is a popular Javascript framework, open sourced by Facebook. The ease of learning React is a key advantage that it has.
+If your project involves many components with different, often changing states , dynamic inputs, user login and access permissions – then the project may be a good fit for React. React helps you manage those changing states and dynamically present different views to the user based on state information.
+This <a href="https://www.intertech.com/Blog/react-tutorial-when-to-use-react-and-why/">article</a> lists the benefits of using React. 
+Similarly, Vue offers us some advantages, like flexibility, simplistic structure and ease of integration.
+
+Every framework has its own pros and cons, meaning it all depends on the developers skills, requirements of the application and many factors like scalability, maintenance, familiarity etc. There are many great tools for developers to choose. Below are a few resources to help you get a brief introduction to these frameworks/libraries.
+
+1. [React vs Angular vs Vue](https://medium.com/@TechMagic/reactjs-vs-angular5-vs-vue-js-what-to-choose-in-2018-b91e028fa91d) - a brief comparison
+2. [Why use Vue](https://medium.com/@brainmobi/advantages-of-using-vue-js-for-your-web-applications-7e460cadfffc) - Advantages of using Vue
+3. [Angular vs React](https://programmingwithmosh.com/react/react-vs-angular/) - A comprehensive comparison between React and Angular
+
+### How to get started with Angular?
+
+1. The official [Angular](https://angular.io/guide/quickstart) website offers a good tutorial to get started with the new framework. This guide will show you how to build and run a simple Angular application.
+
+2. This article on [freeCodeCamp](https://medium.freecodecamp.org/want-to-learn-angular-heres-our-free-33-part-course-by-dan-wahlin-fc2ff27ab451) is a good place to start writing simple applications using the features that Angular has to offer.
+
+3. [Here](https://github.com/angular/angular) is a link to the official Angular framework maintained by Google. You could even open a PR and start contributing!
+
+4. You can also start learning [TypeScript](https://www.typescriptlang.org/docs/home.html) and get comfortable with the new language.
```

Comment:
I think Typescript is a must right? So shouldn't it be you should instead of you can?